testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/admin/PermissionsTest.java

```diff
@@ -295,12 +295,12 @@ public void invoke(RealmResource realm) {
         }, Resource.REALM, false, true);
         assertGettersEmpty(clients.get(AdminRoles.QUERY_REALMS).realm(REALM_NAME).toRepresentation());
 
-        // this should throw forbidden as "query-users" role isn't enough
+        // this should pass given that users granted with "query" roles are allowed to access the realm with limited access
         invoke(new Invocation() {
             public void invoke(RealmResource realm) {
                 clients.get(AdminRoles.QUERY_USERS).realm(REALM_NAME).toRepresentation();
             }
-        }, clients.get(AdminRoles.QUERY_USERS), false);
+        }, clients.get(AdminRoles.QUERY_USERS), true);
```

Comment:
I think we need similar test for each of query-* role. Something like this:

````
for (String role : AdminRoles.ALL_QUERY_ROLES) {
    invoke(realm -> clients.get(role).realms().realm(REALM_NAME).toRepresentation(), clients.get(role), true);
}
````

Reply:
Good point. I'll update the PR with your suggestion.